### PR TITLE
docs: how to set up MiniConstellation on Azure

### PR DESCRIPTION
**Pull request by elchead**
### Context

The goal is to make the first use of Constellation easier for people who DON'T already have a Linux machine.
The easiest setup is MiniConstellation since it does not involve any creation of IAM resources or resource quotas to spin up a full cluster.

As of now MiniConstellation is only supported on Linux. We cannot expect everyone to use Linux locally, especially product managers or other higher level people.

We don't document AWS, since their supported bare-metal machines are very big and expensive.

### Proposed change(s)

- document how to set up Azure to run MiniConstellation

### Related issue
- [AB#3201](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/3201)

### Checklist

- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: docs/docs/getting-started/first-steps-local.md [hunk 1]
```diff
@@ -2,32 +2,101 @@
...
+* machine requirements:
   * An x86-64 CPU with at least 4 cores (6 cores are recommended)
   * At least 4 GB RAM (6 GB are recommended)
   * 20 GB of free disk space
-  * Hardware virtualization enabled in the BIOS/UEFI (often referred to as Intel VT-x or AMD-V/SVM)
-  * `iptables` rules configured to not drop forwarded packages.
-    If running the following command returns no error, please follow [the troubleshooting guide](#vms-have-no-internet-access):
+  * Hardware virtualization enabled in the BIOS/UEFI (often referred to as Intel VT-x or AMD-V/SVM) / nested-virtualization support when using a VM
+* OS / library requirements:
+  * recommended: Ubuntu 22.04 LTS
```
Comment (m1ghtym0) [inline docs/docs/getting-started/first-steps-local.md:72]: Are those requirements below automatically installed with Ubuntu 22.04?

**Review by m1ghtym0 — COMMENTED** [review]
Thanks for building all that Terraform tooling:-)
I'm unsure if `setup-miniconstellation.sh` should be hidden behind a script.
Users might have the experience of executing a bunch of scripts but don't really know what's going on.
Ultimately, it's probably all about interacting with Constellation so it might be fine.

File: docs/docs/getting-started/first-steps-local.md (hunk 1 above)
Comment (elchead) [reply docs/docs/getting-started/first-steps-local.md:72]: I agree it's confusing. Maybe we just limit ourselves to supporting Ubuntu and list all packages that are required (see dev-docs/miniconstellation/azure-terraform/cloud-init.yaml)?

Comment (elchead) [reply docs/docs/getting-started/first-steps-local.md:72]: I agree that the part about using constellation in `setup-miniconstellation.sh` should not be hidden. So I put that in the  first-steps and just refer to the script as a quick way?

**Comment by elchead** [comment]
When you follow the Terraform route, let me know if you had to start the docker service manually, since I had to manually start it on one test machine

**Comment by elchead** [comment]
Etel and I agreed that it doesn't really make sense to add the quick setup to the official docs. Not even costwise does it make sense to spin up a mini cluster in the cloud vs just spawning a regular cloud cluster. We keep the setup in dev-docs though for whenever someone might still want to use it.

**Review by m1ghtym0 — APPROVED** [review]

**Pull request merged**